### ISSUE TITLE
Form Creator  2.12.2 - Pièce jointe ne suit pas le ticket dans son entité

### ISSUE THREAD

**Issue by SDIS-Vendee** [pluginsGLPI/formcreator#2416]
Bonjour,

j'utilise le plugin depuis quelques temps déjà et je rencontre un souci d'usage. 

Les pièces jointes ajoutées en tant que fichier dans un formulaires ne suivent pas dans le ticket si celui-ci n'est est dans une entité particulière. 

Peu importe l'entité du demandeur, de destination de la cible etc... la pièce jointe semble rester sur l'entité racine et le ticket cible n'y est pas lié (sauf si celui-ci est fait dans l'entité racine)

Par avance merci pour votre aide

Julien

**Comment by btry** [comment]
Bonjour

Vos pièces jointes sont ajoutées via une question de type fichier, ou bien il s'agit d'une image déposée dans une zone de texte ?

**Comment by SDIS-Vendee** [comment]
Bonjour,

c'est une question de type fichier en effet. désolé du manque de précisions.

**Comment by btry** [comment]
Ca va être compliqué à résoudre : un document peut être  rattaché à plusieurs tickets (ou changements, eux mêmes dans des entités différentes. Il faut développer un système qui gère ce genre de situations; probablement en dupliquant les documents pour en avoir autant de copies que nécessaire, dans chaque entité des cibles concernées.

**Comment by SDIS-Vendee** [comment]
Je vois la problématique. Par contre, je m'étonne un peu, on ne doit pas être les seuls à utiliser les entités et les formulaires...
Ou alors, il y a quelque chose que je rate quelque part.

**Comment by btry** [comment]
Non, vous n'êtes pas les seuls, des tickets; mais il est tout à fait possible de configurer (spontanément) les formulaires et cibles de manière à dissimuler le problème.Je pense même que c'est le cas d'utilisation le plus courant.

**Comment by SDIS-Vendee** [comment]
Pour information, je suis repassé en version 2.11.2 et l'ajout des PJ dans le ticket cible fonctionne de nouveau quel que soit les entités.
A mon sens il s'agit d'une régression fonctionnelle entre la 2.11 et la 2.12

**Comment by SDIS-Vendee** [comment]
Je viens de me rendre compte que sur cette version les tickets ne sont pas accessibles pour les agents.. Je suis un peu bloqué dans l'utilisation de formcreator actuellement

**Comment by btry** [comment]
Bonjour

En supposant que vos formulaires ne changent pas entre la version 2.11.2 et 2.12.x, avez vous observé que les documents sont placés dans des entités différentes d'une version à l'autre ?

Ce serait bien de décrire votre arborescence d'entités, dans laquelle se trouve le formulaire, dans laquelle se trouve le demandeur qui remplit une demande, et dans laquelle se trouve le document uploadé pour chaque version.

**Comment by SDIS-Vendee** [comment]
Bonjour,
les formulaires sont effectivement les mêmes. 

Les entités sont organisées de la manière suivante : 1 entité racine et 2 sous entités

ENTITE 1 
-->ENTITE 1-1
-->ENTITE 1-2

/////////////////

Les utilisateurs "agents" sont tous sur l'entité "ENTITE 1" en recursif.

Les formulaires sont soit sur l'entité "ENTITE 1-1" ou "ENTITE 1-2" les tickets cibles sont créés sur les entités des formulaires.

//////////////

2 cas d'usage : 
en version 2.11 et 2.10 -> Les tickets ne sont pas accessibles aux agents via le menu "Mes demandes d'assistance" et les compteurs affichent "N/A" en haut sur l'interface "agent" formcreator. PAr contre, l’ajout des pièces jointes fonctionne, elles sont bien rattachées aux tickets cibles.
en version 2.12.X -> Les tickets redeviennent accessibles aux agents et les compteurs sont ok en haut. Par contre, les pièces jointes ne sont plus rattachées aux tickets cibles quelque soit les entités ...

J'ai même effectué des tests en mettant les formulaires dans l'ENTITE 1 + création du ticket dans l'entité ENTITE 1 ... Les pièces jointes ne sont pas rattachées.

**Comment by btry** [comment]
Si les compteurs affichent N/A, c'est qu'il y a une erreur de requête SQL. Des problemes ont existé et sont normalement résolus avec la version 2.11.4 . Si vous avez rétrogradé Formcreator, vous devez avoir restauré un backup de la BDD dans le même temps, sinon le plugin ne fonctionnera pas. Je mets les problèmes de downgrade de coté, car c'est hors sujet.

Je tenterai de reproduire à partir de ce que vous avez décrit.

**Comment by SDIS-Vendee** [comment]
Ok, merci. Je note pour le downgrade. 

Pour information, peu importe la configuration, j'ai fait des essais dans tous les sens la pièce jointe ne semble pas suivre sur le ticket. Elle est bien liée avec le formulaire mais ne suite pas sur le ticket.

**Comment by SDIS-Vendee** [comment]
Bonjour,

petite nouveauté après quelques essais et tests divers, dans mon ticket cible, lorsque j'ajoute la réponse à la question lié à mon upload de fichier dans la description, cela fonctionne correctement. ajout du champ de type ##answer_236## dans la description du ticket... 

Par contre, dans le ticket est ensuite affiché le libellé "document joint"

C'est peut être le fonctionnement normal ou souhaité, dans ce cas autant pour moi, je suis passé à côté de l'info.

**Comment by btry** [comment]
Bonjour

Il est nécessaire de mentionner le tag d'une réponse de type fichier pour que le document soit relié à un ticket généré. Mais cela n'a pas d'influence sur l'entité rattachée au document représentant le fichier uploadé.

C'est le comportement normal. Par contre il y a quelques mois j'ai découvert que c'était cassé. J'ignore depuis quand. Il peut y avoir un lien avec votre problème. Cependant, le choix de l'entité d'un document créé par une question de type fichier reste un point d'amélioration du plugin.
